bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/ForecastSolarObject.java

```diff
@@ -0,0 +1,290 @@
+            if (f.getKey().toLocalDate().equals(queryDateTime.toLocalDate())) {
+                // floor has valid date
+                return f.getValue() / 1000.0;
+            } else {
+                // floor date doesn't fit
+                return UNDEF;
+            }
+        } else if (f == null && c != null) {
+            if (c.getKey().toLocalDate().equals(queryDateTime.toLocalDate())) {
+                // only ceiling from correct date available - no valid data reached yet
+                return 0;
+            } else {
+                // ceiling date doesn't fit
+                return UNDEF;
+            }
+        } else if (f != null && c != null) {
+            // we're during suntime!
+            double powerCeiling = c.getValue();
+            double powerFloor = f.getValue();
+            // calculate in minutes from floor to now, e.g. 20 minutes
+            // => take 2/3 of floor and 1/3 of ceiling
+            double interpolation = (queryDateTime.getMinute() - f.getKey().getMinute()) / 60.0;
+            actualPowerValue = ((1 - interpolation) * powerFloor) + (interpolation * powerCeiling);
+            return actualPowerValue / 1000.0;
+        } // else both null - this shall not happen
+        return UNDEF;
+    }
+
+    public double getDayTotal(LocalDate queryDate) {
+        if (rawData.isEmpty()) {
+            return UNDEF;
+        }
+        JSONObject contentJson = new JSONObject(rawData.get());
+        JSONObject resultJson = contentJson.getJSONObject("result");
+        JSONObject wattsDay = resultJson.getJSONObject("watt_hours_day");
+
+        if (wattsDay.has(queryDate.toString())) {
+            return wattsDay.getDouble(queryDate.toString()) / 1000.0;
+        }
+        return UNDEF;
+    }
+
+    public double getRemainingProduction(ZonedDateTime queryDateTime) {
+        if (wattHourMap.isEmpty()) {
+            return UNDEF;
+        }
+        double daily = getDayTotal(queryDateTime.toLocalDate());
+        double actual = getActualValue(queryDateTime);
+        if (daily < 0 || actual < 0) {
+            return UNDEF;
+        }
+        return daily - actual;
+    }
+
+    public ZoneId getZone() {
+        return zone;
+    }
+
+    @Override
+    public String toString() {
+        return "Expiration: " + expirationDateTime + ", Valid: " + valid + ", Data:" + wattHourMap;
+    }
+
+    /**
+     * SolarForecast Interface
+     */
+    @Override
+    public State getDay(LocalDate localDate, String... args) {
+        if (args.length > 0) {
+            logger.info("ForecastSolar doesn't accept arguments");
+            return UnDefType.UNDEF;
+        }
+        double measure = getDayTotal(localDate);
+        return Utils.getEnergyState(measure);
+    }
+
+    @Override
+    public State getEnergy(Instant start, Instant end, String... args) {
+        if (args.length > 0) {
+            logger.info("ForecastSolar doesn't accept arguments");
+            return UnDefType.UNDEF;
```

Comment:
Is this expected to happen? Perhaps you could instead throw `IllegalArgumentException` to fail fast?